### DISCUSSION TITLE
Eddy (from FSL) in NeuroDesktop

### DISCUSSION THREAD

**Discussion by mschira** [neurodesk/neurodesk.github.io#104] [Q&A] [answered]
Hi
I am currently exploring the NeuroDesktop running on Nectar.
Indeed I am just tipping my toe into the water.
So first up I wanted to try doing some DWI data preprocessing.
So I opened a shell, loaded MRtrix and FSL
>ml fsl/6.0.4
>ml mrtrix3/3.0.3

Then I tried 
>eddy
and I got command not found, where I should have gotten the usage instructions for eddy.

Any ideas why that might be?
Cheers
Mark

**Comment by stebo85** [comment, marked as answer]
Dear @mschira,

Wonderful to hear that you are exploring neurodesk :)

When you have a look at FSL's eddy manual: https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/eddy/UsersGuide#The_eddy_executables

it describes how to start eddy:

<img width="1046" alt="image" src="https://user-images.githubusercontent.com/4021595/186810608-4beeeaee-44a7-4fc1-8f60-7e83ec98a863.png">

the executable "eddy" does not exist in FSL (maybe it used to be called eddy in older versions?).

in order to run eddy in neurodesk, the best way right now is to use the openmp version (because nectar does not yet offer GPUs for desktop sessions): "eddy_openmp"

<img width="713" alt="image" src="https://user-images.githubusercontent.com/4021595/186810843-a0c291b2-1b95-4534-8749-8c9aec56c512.png">

Thank you 
Kind regards und viele Gruesse
Steffen

**Comment by mschira** [comment]
Oh, wow, that must be a version difference.
Thanks for pointing that out.
<img width="983" alt="Screen Shot 2022-08-26 at 14 04 20" src="https://user-images.githubusercontent.com/9832335/186818285-f795ef2d-f76a-4d11-a94a-e248fb234a87.png">

I don't have eddy_openmp on my computer. 
Thanks
M.

**Reply by stebo85** [reply]
Yes, that's very interesting. I could create a soft link to eddy so people don't run into this?

**Comment by mschira** [comment]
Tricky I suppose, dammed if you do (cause trouble somewhere else?) dammed if you don't.

I investigated a bit further, and the computer I regularly type >eddy is a Mac, and in my use case, I never RUN eddy on a Mac, I just call >eddy to look up the usage and then do the actual running of eddy on a linux machine with an Nvidia GPU. 
Assuming that this peculiarity of eddy as opposed to eddy_openmp and eddy_cuda9.1 is a Mac issue, due to their lack of support for CUDA.

So I guess in the light of all this, maybe don't?
Are you guys working with the MRTrix team in any shape or form? I can imagine having their input on how to optimally configure these containers is useful?
